### PR TITLE
fix:  handling kwargs list and handling kms keys for serial inference pipeline.

### PR DESCRIPTION
**Pull request by sreedes**
*Issue #, if available:*
1. The keyword arguments for source_dir, dependencies need to be popped, otherwise it is going to be duplicated for the repack model step, if the user also passes in those parameters.
2. The model_kms_key needs to be handled in case of serial inference pipeline as well.

*Description of changes:*

*Testing done:*

## Merge Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your pull request._

#### General

- [x] I have read the [CONTRIBUTING](https://github.com/aws/sagemaker-python-sdk/blob/master/CONTRIBUTING.md) doc
- [x] I used the commit message format described in [CONTRIBUTING](https://github.com/aws/sagemaker-python-sdk/blob/master/CONTRIBUTING.md#committing-your-change)
- [ ] I have passed the region in to all S3 and STS clients that I've initialized as part of this change.
- [ ] I have updated any necessary documentation, including [READMEs](https://github.com/aws/sagemaker-python-sdk/blob/master/README.rst) and [API docs](https://github.com/aws/sagemaker-python-sdk/tree/master/doc) (if appropriate)

#### Tests

- [ ] I have added tests that prove my fix is effective or that my feature works (if appropriate)
- [ ] I have checked that my tests are not configured for a specific region or account (if appropriate)
- [ ] I have used [`unique_name_from_base`](https://github.com/aws/sagemaker-python-sdk/blob/master/src/sagemaker/utils.py#L77) to create resource names in integ tests (if appropriate)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Review by EthanShouhanCheng — APPROVED** [review]
LGTM

**Review by jerrypeng7773 — APPROVED** [review]

File: src/sagemaker/workflow/step_collections.py [hunk 1]
```diff
@@ -130,13 +130,10 @@ def __init__(
...
-        # remove kwargs consumed by model repacking step
-        kwargs.pop("entry_point", None)
-        kwargs.pop("source_dir", None)
-        kwargs.pop("dependencies", None)
-        kwargs.pop("output_kms_key", None)
+            # remove kwargs consumed by model repacking step
+            kwargs.pop("output_kms_key", None)
 
-        if model is not None:
+        elif model is not None:
```
Comment (shreyapandit) [inline src/sagemaker/workflow/step_collections.py:136]: Hi @sreedes, can you clarify why this change to an `elif` is necessary? It seems like checking for existence of `entry_point` and `model` are two independent things, and we would likely want to check both?

**Review by shreyapandit — CHANGES_REQUESTED** [review]

File: src/sagemaker/workflow/step_collections.py (hunk 1 above)
Comment (sreedes) [reply src/sagemaker/workflow/step_collections.py:136]: Both of them cannot appear together, the customer could either send it through model or directly through RegisterModelStep(existing mechanism), we would recommend them to use it through Model, since it comes packed with framework information as well, hence the else.

Comment (shreyapandit) [reply src/sagemaker/workflow/step_collections.py:136]: Thanks for clarifying!

**Review by shreyapandit — APPROVED** [review]
